src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        GeometricFigure figure = new GeometricFigureSupplier().getRandomFigure();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        if (figure == GeometricFigure.CIRCLE) {
+            return new Circle(random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.RECTANGLE) {
+            return new Rectangle(random.nextInt(), random.nextInt(),
+                    colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.ISOSCELES_TRAPEZOID) {
+            return new IsoscelesTrapezoid(random.nextInt(), random.nextInt(),
+                    random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.SQUARE) {
+            return new Square(random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        return new RightTriangle(random.nextInt(), random.nextInt(),
+                colorSupplier.getRandomColor().toLowerCase());
```

Comment:
Don't return 'null' from a method. If you have a default case that should be handled, it's better to include it in the 'default' case of a switch statement rather than having a return outside of the if-else blocks.